src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -88,6 +88,14 @@ def query(self, filters, options):
                 pools = [i.__getstate__(**state_kwargs) for i in zfs.pools]
         return filter_list(pools, filters, options)
 
+    def query_imported_fast(self):
+        # the equivalent of running `zpool list -H -o guid,name` from cli
+        try:
+            with libzfs.ZFS() as zfs:
+                return {str(i.guid): i.name for i in zfs.pools}
+        except libzfs.ZFSException as e:
+            raise CallError(f'Failed listing imported pools with error: {e}')
```

Comment:
Let's reword the error here please and remove the `imported` wording

Reply:
No. Why would I remove that word? It's, literally, trying to list imported pools.